### ISSUE TITLE
Out of memory exception for large TTL file

### ISSUE THREAD

**Issue by PavelPZ** [mdesalvo/RDFSharp#59]
Do you have a plan to allow parsing large TTL files?

I have 64 GByte RAM and try to read 1.2 GByte TTL. Exception raised when RDFSharp tries
to load file into memory string.

I can modify a file to fit into memory (by replacing long identifiers). I do not know if the problem occurs later on parsing (because of eg. 2 GByte limit for List, etc.).

**Comment by mdesalvo** [comment]
Dear Pavel,
thanks for reporting this performance issue.

It is most probably caused by the way the TTL file is loaded into memory before starting parse: this is a bottleneck for use cases requiring consume of very large amounts of data and needs optimization, which is under consideration but not in scope of the current 2.x branch (implementation of SPARQL 1.1 features).

The workaround you identified is correct: shorting the Uris by registering the most used namespaces can help you achieving the goal. Or also splitting the file into two smaller files and then merging their graphs.

I would really appreciate if you could contribute to the project by designing a solution to this issue, you are welcome to code and ideas submission!

Regards,
Marco
